File: app/src/main/java/com/foxek/simpletimer/di/component/ActivityComponent.java
package com.foxek.simpletimer.di.component;

import com.foxek.simpletimer.di.PerActivity;
import com.foxek.simpletimer.di.module.ActivityModule;
import com.foxek.simpletimer.ui.interval.IntervalActivity;
import com.foxek.simpletimer.ui.interval.dialog.IntervalCreateDialog;
import com.foxek.simpletimer.ui.interval.dialog.IntervalEditDialog;
import com.foxek.simpletimer.ui.interval.dialog.WorkoutEditDialog;
import com.foxek.simpletimer.ui.timer.TimerActivity;
import com.foxek.simpletimer.ui.workout.WorkoutActivity;
import com.foxek.simpletimer.ui.workout.dialog.WorkoutCreateDialog;

import dagger.Component;

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(WorkoutActivity activity);

    void inject(IntervalActivity activity);

    void inject(TimerActivity activity);

    void inject(WorkoutCreateDialog dialog);

    void inject(WorkoutEditDialog dialog);

    void inject(IntervalCreateDialog dialog);

    void inject(IntervalEditDialog dialog);

}

File: app/src/main/java/com/foxek/simpletimer/di/component/ActivityComponent.kt
package com.foxek.simpletimer.di.component

import com.foxek.simpletimer.di.PerFragment
import com.foxek.simpletimer.di.module.ActivityModule
import com.foxek.simpletimer.presentation.editworkout.EditWorkoutFragment
import com.foxek.simpletimer.presentation.round.RoundFragment
import com.foxek.simpletimer.presentation.round.dialog.RoundCreateDialog
import com.foxek.simpletimer.presentation.round.dialog.RoundEditDialog
import com.foxek.simpletimer.presentation.workout.WorkoutFragment
import com.foxek.simpletimer.presentation.workout.dialog.WorkoutCreateDialog

import dagger.Component

@PerFragment
@Component(dependencies = [ApplicationComponent::class], modules = [ActivityModule::class])
interface ActivityComponent {

    fun inject(fragment: WorkoutFragment)

    fun inject(fragment: RoundFragment)

    fun inject(fragment: EditWorkoutFragment)

    fun inject(dialog: WorkoutCreateDialog)

    fun inject(dialog: RoundCreateDialog)

    fun inject(dialog: RoundEditDialog)
}

